packages/lit-dev-content/site/blog/2023-10-10-lit-3.0.md

```diff
@@ -0,0 +1,175 @@
+[`@lit/context`](https://www.npmjs.com/package/@lit/context)
+
+These packages can now be considered stable, and have documentation and examples on lit.dev!
+
+Current users of these packages under the `@lit-labs` scope can migrate by first updating to the latest version of those packages to test for any breakages, then update the import to use the `@lit` scoped package. The latest `@lit-labs` scoped packages will receive the same updates until the next major version of the `@lit` scoped package.
+
+Thank you so much to everyone in the community who has tested out a Labs package, filed issues, discussed features, and led these packages to graduation! 🎓
+
+### React
+
+While custom elements can be used in React projects as is, there are some rough edges around their usage, namely:
+- Setting properties on elements (rather than attributes)
+- Adding handlers for custom events
+- Type checking custom elements and their props
+
+Some of these are being addressed by React in a future version, but they are currently only present in experimental builds not recommended for production.
+
+`@lit/react` allows creation of React components wrapping the custom element. The created components can be idiomatic to React such that users of the component do not have to worry about the inner web component implementaion.
+
+It is useful for both web component authors who wish to vend React versions of their components for users to reach a wider audience, as well as React developers who wish to use a neat web component that they found in their project more ergonomically.
+
+`@lit-labs/react` has been our most popular labs project by far with over 500k weekly npm downloads and is already being used by many web component libraries to provide React versions of their components to users.
+
+Read more about it at our [React framework integration](/docs/frameworks/react) documentation.
+
+### Task
+
+Rendering asynchronously retrieved data in a Lit component, while not terribly difficult to do, can involve a lot of boiler plate code to do properly with all the edge cases to handle.
+
+Some considerations include:
+- Conditionally rendering different content while the data is pending, retrieved, or the request failed
+- Handling race conditions, ensuring the data for the latest request is kept and previous requests are cancelled or ignored if completed later
+
+`@lit/task` provides a reactive controller to help manage this async workflow.
+
+We've found ourselves, on numerous occasions, starting a barebones example with asynchronous requests and quickly reaching to `@lit-labs/task` for its convenience. We've refined the API and added features like providing an `AbortSignal` while the project was being developed under labs. We are happy to officially graduate and encourage everyone to use it!
+
+Checkout the details with examples and various usages in the [Async Tasks](/docs/data/task/) documentation.
+
+### Context
+
```

Comment:
```suggestion
`@lit/context` is an interoperable system for an element to request data from any of its ancestors. It's useful for shared elements to receive configuration from their environemnt, for elements in a composable plugin system to exhange data, and as an interoperable DOM-based dependency injection system.

Start by creating a context using the `createContext()` function. A context acts as a key, for the consumer and provider of the data to identify each other.

Then, on the prividing element, declare a property with the `@provide` decorator. As with the main `lit` decorators, these work as standard decorators, and with TypeScript's `experimentalDecorators` option enabled. The property's value will be made available to any descendents of the element that consume the context.

Finally, on the consuming element, declare a property with the `@consume` decorator. When the element attaches to the DOM it will receive the current value of the first ancestor that provides the context. If called with `subscribe: true` then it will also receive updates as the value changes.

For more info, see the [Context](/docs/data/context/) documentation.
```